app/helpers/cdx_form_helper.rb

```diff
@@ -79,53 +79,80 @@ def form_actions(&block)
   end
 
   def fields_for(record_name, record_object = nil, fields_options = {}, &block)
-    if record_object.nil? && @object.respond_to?(record_name)
-      record_object = @object.send(record_name)
-    end
-
-    super(record_name, record_object, objectify_options(fields_options)) do |form|
-      # Clear errors that are handeld in the nested form
-      form.errors_to_show.each do |nested_key|
-        errors_to_show.delete("#{record_name}.#{nested_key}".to_sym)
-      end
-
+    fields_options[:nested_errors_namespace] = prefixed_error_key(record_name)
+    super(record_name, record_object , objectify_options(fields_options)) do |form|
+      form.errors_to_show = errors_to_show
       block.call(form)
-
       @template.concat form.form_errors
     end
   end
 
   def has_error?(attr_name)
-    errors_to_show.include?(attr_name.to_sym)
+    errors_to_show.include?(prefixed_error_key(attr_name))
+  end
+
+  # Returns `true` if the object has any errors.
+  def has_errors?
+    !object_errors.empty?
   end
 
   protected
 
-  def errors_to_show
-    @errors_to_show ||= @object.errors.keys
+  def errors_to_show=(errors_to_show)
+    @errors_to_show = errors_to_show
   end
 
   private
 
+  def errors_to_show
+    @errors_to_show ||= object_errors.keys
+  end
+
+  def object_errors
+    (@object || @options[:object]).errors
+  end
+
+  def prefixed_error_key(key)
+    if nested_prefix = @options[:nested_errors_namespace]
+      "#{nested_prefix}.#{key}"
+    else
+      key
+    end.to_sym
+  end
+
   def error_messages_to_show
+    nested_prefix = @options[:nested_errors_namespace]
+    nested_prefix = "#{nested_prefix}." if nested_prefix
     Hash[errors_to_show.map do |attribute|
-      messages = @object.errors.full_messages_for(attribute)
+      next if nested_prefix && !attribute.to_s.starts_with?(nested_prefix)
+
+      messages = object_errors.full_messages_for(attribute)
       [attribute, messages] unless messages.empty?
     end.compact]
   end
 
+  def clear_errors_to_show
+    if nested_prefix = @options[:nested_errors_namespace]
+      nested_prefix = "#{nested_prefix}."
+      errors_to_show.delete_if { |key| key.to_s.starts_with?(nested_prefix) }
+    else
+      errors_to_show.clear
+    end
+  end
+
   def log_unhandled_errors
     unhandled_errors = error_messages_to_show.reject { |k, _| k == :base }
     return if unhandled_errors.empty?
 
+    message = "Unhandled form errors in #{object_name}: #{unhandled_errors}"
     if Rails.env.test?
-      raise "Unhandled form errors in #{@object.model_name}: #{unhandled_errors}"
+      raise message
     end
 
-    Rails.logger.info "Unhandled form errors in #{@object.model_name}: #{unhandled_errors}"
-    Raven.capture_message("Unhandled form errors", extra: {
-      form: @object.model_name,
+    Rails.logger.info message
+    Raven.capture_message("Unhandled form errors",
+      form: object_name,
       errors: unhandled_errors,
-    })
+    )
```

Comment:
I don't know how the new Sentry gem is working, but I just verified the legacy Raven source code, and we must specify the `extra:` key (see `Raven::Event.new`).

```ruby
    Raven.capture_message("Unhandled form errors", extra: {
      form: object_name,
      errors: unhandled_errors,
    })
```